tests/PKSim.Tests/IntegrationTests/SimulationWithParticlesFormulationSpecs.cs

```diff
@@ -0,0 +1,496 @@
+      {
+         var compoundName = _compound.Name;
+
+         var lumenPaths = (from segment in _lumenSegments
+                           select $"Organism|Lumen|{segment}|{compoundName}").ToArray();
+
+         var binPaths = new string[numberOfBins];
+         var binSolidDrugPaths = new string[numberOfBins];
+         var binInsolubleDrugPaths = new string[numberOfBins];
+         var binSolidDrugPerSegmentPaths = new string[numberOfBins][];
+         var binParticlesFractionPerSegmentPaths = new string[numberOfBins][];
+
+         for (int binIdx = 0; binIdx < numberOfBins; binIdx++)
+         {
+            binPaths[binIdx] = binPathFor(binIdx);
+            binSolidDrugPaths[binIdx] = binSolidDrugPathFor(binPaths[binIdx], compoundName);
+            binInsolubleDrugPaths[binIdx] = binInsolubleDrugPathFor(binPaths[binIdx], compoundName);
+            binSolidDrugPerSegmentPaths[binIdx] = binSolidDrugPerSegmentPathsFor(binSolidDrugPaths[binIdx]);
+            binParticlesFractionPerSegmentPaths[binIdx] = binParticlesFractionPerSegmentPathsFor(binSolidDrugPaths[binIdx]);
+         }
+
+         return (lumenPaths, binSolidDrugPaths, binInsolubleDrugPaths, 
+                 binSolidDrugPerSegmentPaths, binParticlesFractionPerSegmentPaths);
+      }
+
+      private string applicationPath => $"Applications|{_protocol.Name}|{_formulation.Name}|Application_1";
+
+      private string binPathFor(int binNumber) => $"{applicationPath}|ParticleBin_{binNumber+1}";
+
+      private string binSolidDrugPathFor(string binPath, string drugName) => $"{binPath}|{drugName}";
+
+      private string binInsolubleDrugPathFor(string binPath, string drugName) => $"{binPath}|InsolubleDrug|{drugName}";
+
+      private string[] binSolidDrugPerSegmentPathsFor(string binSolidDrugPath) => binParameterPerSegmentPaths(binSolidDrugPath, "Solid drug");
+
+      private string[] binParticlesFractionPerSegmentPathsFor(string binSolidDrugPath)=> binParameterPerSegmentPaths(binSolidDrugPath, "Number of particles fraction");
+
+      private string[] binParameterPerSegmentPaths(string binSolidDrugPath, string parameter)
+      {
+         return (from segment in _lumenSegments
+                 select $"{binSolidDrugPath}|ParticlesDrugRelease{segment}|{parameter}").ToArray();
+      }
+
+      private IContainer moleculeProperties => _simulation.Model.Root.Container(_compound.Name);
+
+      /// <summary>
+      /// for debug purposes only
+      /// </summary>
+      protected void ExportSimulation()
+      {
+         var exporter = IoC.Resolve<IMoBiExportTask>();
+         exporter.SaveSimulationToFile(_simulation, @"C:\Temp\IntegrationTestSim.pkml");
+      }
+
+      //--------------------------------------------------------------------------------------------
+      // Test routins called by observations in all derived classes
+      //--------------------------------------------------------------------------------------------
+
```

Comment:
protected void CheckXXX() routines are the essential checks and are called from all the test classes below.

More tests will follow next week